### PR TITLE
Add `H-optimus-0` model to registry

### LINKED ISSUE #583: Onboard `H-optimus-0` model
Bioptimus recently released the `H-optimus-0` model:
https://github.com/bioptimus/releases/tree/main/models/h-optimus/v0

We should onboard this model, run evaluations and add it to our leaderboard.

### PR DESCRIPTION
**Pull request by nkaenzig**
Closes #583

### CODE REVIEW

File: docs/user-guide/advanced/replicate_evaluations.md [hunk 1]
```diff
@@ -140,6 +137,18 @@ IN_FEATURES=1024 \
 eva predict_fit --config configs/vision/pathology/offline/<task>.yaml
 ```
 
+### H-optimus-0 (Bioptimus) - ViT-G14 [[5]](#references)
+Bioptimus released their H-optimus-0 which was trained on a collection of 500,000 H&E slides. The model weights
```
Comment (roman807) [inline docs/user-guide/advanced/replicate_evaluations.md:141]: link `Bioptimus` to https://www.bioptimus.com/?

File: src/eva/vision/models/networks/backbones/pathology/bioptimus.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+from eva.vision.models.networks.backbones.registry import register_model
+
+
+@register_model("pathology/bioptimus_h_optimus_0")
+def bioptimus_h_optimus_0(
+    dynamic_img_size: bool = True,
+    out_indices: int | Tuple[int, ...] | None = None,
+) -> nn.Module:
+    """Initializes the h_optimus_0 pathology FM by bioptimus.
```
Comment (roman807) [inline src/eva/vision/models/networks/backbones/pathology/bioptimus.py:16]: nit: `Bioptimus`

**Review by roman807 — APPROVED** [review]

File: docs/user-guide/advanced/replicate_evaluations.md (hunk 1 above)
Comment (nkaenzig) [reply docs/user-guide/advanced/replicate_evaluations.md]: done

File: src/eva/vision/models/networks/backbones/pathology/bioptimus.py (hunk 2 above)
Comment (nkaenzig) [reply src/eva/vision/models/networks/backbones/pathology/bioptimus.py]: done

**Pull request merged**